cogs/welcome/welcome.py

```diff
@@ -318,6 +343,74 @@ async def logServerLeave(self, leaveUser: discord.Member):
                     leaveUser.id,
                 )
 
+    async def generateRandWelcomeImg(self, user: discord.member, guild: discord.guild):
+        """create an image for the specific player using their avatar and an image from the random image pool, then returns it"""
+        base = Image.open(
+            self.imgDir
+            / str(guild.id)
+            / random.choice(os.listdir(os.path.join(self.imgDir, str(guild.id))))
+        )
+        mask = Image.open(os.path.join(data_manager.bundled_data_path(self), "MASK.png"))
+        borderOverlay = Image.open(
+            os.path.join(data_manager.bundled_data_path(self), "BORDER.png")
+        )
+        borderOverlayMask = Image.open(
+            os.path.join(data_manager.bundled_data_path(self), "BORDER_mask.png")
+        )
+        # get avatar from User
+        avatar: bytes
+        retrievedAvatar: Image
+        session = aiohttp.ClientSession()
+        # a header to successfully download user avatars for use
+        usedHeader = {"User-agent": "Mozilla/5.0"}
+
+        try:
+            async with session.get(str(user.avatar.url), headers=usedHeader) as webp:
+                avatar = await webp.read()
+                retrievedAvatar = Image.open(io.BytesIO(avatar))
+                if not retrievedAvatar:
+                    base.close()
+                    mask.close()
+                    borderOverlay.close()
+                    borderOverlayMask.close()
+                    LOGGER.error(
+                        "Could not retrieve user profile picture from discord servers.", exc_info=1
+                    )
+                    return
+        except aiohttp.ClientResponseError:
+            pass
+
+        retrievedAvatar = retrievedAvatar.resize((325, 325), 1)
+        base.paste(borderOverlay, (434, 0), borderOverlayMask)
+        base.paste(retrievedAvatar, (434, 0), mask)
+        generated = io.BytesIO()
+        base.save(generated, format="png")
+        generated.seek(0)
+        base.close()
+        mask.close()
+        borderOverlay.close()
+        borderOverlayMask.close()
+        return generated
+
+    async def ensureCurrentServerHasImgCache(self, channel: discord.channel):
+        """
+        Check if there is a folder in the image cache for the associated server. If one doesn't exist, creates it.
+        """
+        idStr = str(channel.guild.id)
+        fp = os.path.join(self.imgDir, idStr)
```

Comment:
`self.imgDir` is a `pathlib.Path` instance. Can make use of it instead of `os.path` functions.